docs/dev_guide/software-required.md

```diff
@@ -0,0 +1,55 @@
+# Get the required software for Linux and MacOS
+Here we will explain the software you need to use Linux or MacOS for `kind`
+development.
+Before you being you must have:
+
+* GitHub account
```

Comment:
we can probably just link to github's page for this without having a guide for getting a github account.

Reply:
Changed the "get a github account step" to "read the k8s community guidelines." Tell me what you think